phoenix-core/src/main/java/org/apache/phoenix/util/JDBCUtil.java

```diff
@@ -219,7 +221,10 @@ public static String formatZookeeperUrl(String zkUrl){
         Preconditions.checkArgument(hosts.length > 0,"Unexpected zk url format no hosts found.");
         String hostsStrings = Arrays.stream(hosts).sorted().collect(Collectors.joining(","));
         components[0] = hostsStrings;
-        return Arrays.stream(components).collect(Collectors.joining(":"));
+        // host:port:path:principal
+        // additional arguments passed in url, strip them out
+        int endIdx = Integer.min(components.length, 3);
```

Comment:
nit: i wonder if we should also make initial validation check for len: 3 or 4
```
Preconditions.checkArgument(components.length == 3 || components.length == 4, "Unexpected zk url format.");
```

Reply:
Actually, the number of components can be greater than 4 also